### ISSUE TITLE
How to match empty ProtocolNames?

### ISSUE THREAD

**Issue by PaulPawletta** [Donders-Institute/bidscoin#110]
Dear Maintainers,

we are using bidscoin to convert PACS data. Now this data contains many unrelevant DICOM series that can be excluded e.g. Summary reports, PACS reports etc. These series have usually an empty string set for the DICOM ProtocolName attribute.

Hence, I would like to exclude all data that has an empty string ProtocolName. I tried the following 2 options:

- define under **exclude** folder **ProtocolName: ''** -> all data will be ignored, even if the ProtocolName is set to a value
- define under **exclude** folder **ProtocolName: '^(?![\s\S])'** -> this Regex should only match an empty string, however the series data is not excluded

What would be the best way to achieve this?

**Comment by marcelzwiers** [comment]
Mhhhh, the second option should indeed work... And all the other attributes of your exclude run-item are empty?

**Comment by PaulPawletta** [comment]
My example bidsmap.yaml looks like this:
<img width="667" alt="Screenshot 2021-09-29 at 10 52 36" src="https://user-images.githubusercontent.com/16834721/135236007-70118070-e29d-4aa7-8804-0418e29742f8.png">

using this config, bidscoin will map everything to extra_data/

**Comment by marcelzwiers** [comment]
I tried to be smart and in `bids.py` > `match_attribute()` I added:
```
if not attribute or not pattern:
        return False
```
You can either change that to:
```
if not pattern:
        return False
```
or install the latest patched (ef889589e7d1b0a54791849df700688ba0e96ce0) dev code...

**Comment by PaulPawletta** [comment]
@marcelzwiers thanks for the quick fix. It's working now.
